coriolis/tests/osmorphing/netpreserver/test_interfaces.py

```diff
@@ -0,0 +1,149 @@
+            "address 192.168.1.20/24\n"
+        )
+        mock_read_file_sudo.return_value = interfaces_content.encode()
+        mock_test_path.return_value = True
+        mock_exec_cmd_chroot.return_value = ""
+
+        self.netpreserver.interface_info = {}
+        self.netpreserver.parse_network()
+
+        expected_info = {
+            "eth0": {
+                "mac_address": "00:11:22:33:44:55",
+                "ip_addresses": ["192.168.1.10"]
+            },
+            "eth1": {
+                "mac_address": "",
+                "ip_addresses": ["192.168.1.20"]
+            }
+        }
+        self.assertEqual(self.netpreserver.interface_info, expected_info)
+
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_read_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_test_path')
+    def test_parse_network_with_source(self, mock_test_path,
+                                       mock_exec_cmd_chroot,
+                                       mock_read_file_sudo):
+        main_content = (
+            "iface eth0 inet static\n"
+            "hwaddress ether 00:11:22:33:44:55\n"
+            "address 192.168.1.10/24\n"
+            "source /etc/network/interfaces.d/\n"
+        )
+        extra_file = "extra_iface"
+        extra_content = (
+            "iface eth1 inet dhcp\n"
+            "address 192.168.1.20/24\n"
+        )
+
+        mock_test_path.return_value = True
+        mock_exec_cmd_chroot.return_value = extra_file
+        mock_read_file_sudo.side_effect = (
+            main_content.encode(), extra_content.encode()
+        )
+
+        self.netpreserver.parse_network()
+
+        expected_info = {
+            "eth0": {
+                "mac_address": "00:11:22:33:44:55",
+                "ip_addresses": ["192.168.1.10"]
+            },
+            "eth1": {
+                "mac_address": "",
+                "ip_addresses": ["192.168.1.20"]
+            }
+        }
+        self.assertEqual(self.netpreserver.interface_info, expected_info)
+
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_read_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_test_path')
+    def test_parse_network_with_trailing_hwaddress(self, mock_test_path,
+                                                   mock_exec_cmd_chroot,
+                                                   mock_read_file_sudo):
+
+        interfaces_content = (
+            "hwaddress ether 00:11:22:33:44:55\n"
+        )
+        mock_read_file_sudo.return_value = interfaces_content.encode()
+        mock_test_path.return_value = True
+        mock_exec_cmd_chroot.return_value = ""
+
+        with self.assertLogs(level='WARNING') as cm:
+            self.netpreserver.parse_network()
+
+        self.assertTrue(any("Found MAC address 00:11:22:33:44:55 does not "
```

Comment:
We don't care about what the actual message is. This is excessive and bad for when we change. Something as trivial as changing a message shouldn't mean you need to also change the test :D 
Please remove.